src/RequestHandler.php

```diff
@@ -50,9 +50,12 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
     {
         $requestHandler = $request->getAttribute($this->handlerAttribute);
 
-        if (is_string($requestHandler)) {
-            $container = $this->container ?: new RequestHandlerContainer();
-            $requestHandler = $container->get($requestHandler);
+        if (is_string($requestHandler) && $this->container->has($requestHandler)) {
+            $requestHandler = $this->container->get($requestHandler);
+        }
+
+        if (is_callable($requestHandler)) {
+            $requestHandler = new CallableHandler($requestHandler);
```

Comment:
Might as well use existing `instanceof` checks for `CallableHandler`.

Reply:
[`Middlewares\Utils\CallableHandler`](https://github.com/middlewares/utils/blob/master/src/CallableHandler.php) implements the both interfaces (MiddlewareInterface and RequestHandlerInterface) and is callable (has an __invoke too), so I'd move `is_callable()` to the end, in order to avoid a CallableHandler inside other CallableHandler.